diff --git a/node_modules/@strapi/database/dist/index.js b/node_modules/@strapi/database/dist/index.js
index 0443808..fa96734 100644
--- a/node_modules/@strapi/database/dist/index.js
+++ b/node_modules/@strapi/database/dist/index.js
@@ -985,6 +985,7 @@ class SqliteDialect extends Dialect {
   }
 }
 const getDialectClass = (client) => {
+
   switch (client) {
     case "postgres":
       return PostgresDialect;
@@ -997,6 +998,10 @@ const getDialectClass = (client) => {
   }
 };
 const getDialectName = (client) => {
+  if (typeof client.dialect === 'string') {
+    client = client.dialect;
+  }
+
   switch (client) {
     case "postgres":
       return "postgres";
@@ -1004,6 +1009,7 @@ const getDialectName = (client) => {
     case "mysql2":
       return "mysql";
     case "sqlite":
+    case "libsql":
     case "sqlite-legacy":
       return "sqlite";
     default:
